icefoganalytics/internal-data-portal
# Fix Datasets Table Render Crash From Bad Acronym

Fixes https://github.com/icefoganalytics/internal-data-portal/issues/104

# Context

Some of my datasets weren't rendering in the All Datasets page. It turned out this was due do the acronym generator crashing because some user-group names ended with trailing whitespace.

# Implementation

Fix running production build locally.
Set up front-end test suite.
Clean new user group names when they come in.
Clean old user group names via migration.

# Screenshots

![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/708363c5-b405-4751-a457-aa621ebd50e6)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to http://localhost:8080/datasets?perPage=10&page=1, and check that I can now see datasets where Department: `Economic Development` -> Division `Corporate Services`
